src/System.Data.Common/src/System.Data.Common.csproj

```diff
@@ -217,6 +217,7 @@
     <Compile Include="System\Data\Common\DbParameter.cs" />
     <Compile Include="System\Data\Common\DbParameterCollection.cs" />
     <Compile Include="System\Data\Common\DbProviderFactory.cs" />
+    <Compile Include="System\Data\Common\DbProviderFactory.netcoreapp.cs" Condition="'$(TargetsNetCoreApp)' == 'true'" />
```

Comment:
Like in the previous PR, I suspect adding things in a separate *.netcoreapp.cs file is unnecessary. If this is correct, my preference is to consolidate the changes into DbProviderFactory.cs.

Reply:
Confirmed that this is OK - consolidated the changes (except for tests).